### PR TITLE
Add categories for variable products to ec tracking code

### LINKED ISSUE #79: Enhanced Ecommerce showing attributes, not categories for variable products
When turning on Enhanced Ecommerce in GA, the Product Category (Enhanced Ecommerce) report shows product attribute values (e.g. "Size: 10") for some products (dresses), and the product category for others (e.g. bags).

This is likely due to the dresses being variable products with variations based on size. However, the product category should still be used for those.

### PR DESCRIPTION
**Pull request by laurendavissmith**
Fixes #79 

-------------------
- [ ] Make sure your changes respect [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- [ ] Did you make changes, or create a **new .js file**? Make sure you ran `grunt` before creating this Pull Request.

### CODE REVIEW

File: includes/class-wc-google-analytics-js.php [hunk 1]
```diff
@@ -418,6 +418,7 @@ function add_item_enhanced( $order, $item ) {
 		$code .= "'id': '" . esc_js( $_product->get_sku() ? $_product->get_sku() : $_product->get_id() ) . "',";
 		$code .= "'name': '" . esc_js( $item['name'] ) . "',";
 		$code .= "'category': " . self::product_get_category_line( $_product );
+		$code .= "'variant': " . self::product_get_varient_line( $_product );
```
Comment (roykho) [inline includes/class-wc-google-analytics-js.php]: Method name spelling.

File: includes/class-wc-google-analytics-js.php [hunk 2]
```diff
@@ -431,21 +432,40 @@ function add_item_enhanced( $order, $item ) {
...
 		if ( is_array( $variation_data ) && ! empty( $variation_data ) ) {
-			$code = "'" . esc_js( wc_get_formatted_variation( $variation_data, true ) ) . "',";
-		} else {
-			$out = array();
-			$categories = get_the_terms( $_product->get_id(), 'product_cat' );
-			if ( $categories ) {
-				foreach ( $categories as $category ) {
-					$out[] = $category->name;
-				}
+			$parent_product = wc_get_product( $_product->get_parent_id() );
```
Comment (roykho) [inline includes/class-wc-google-analytics-js.php]: `get_parent_id()` method needs a pre WC 3.0 fallback.

**Comment by laurendavissmith** [comment]
@roykho ready for another look

**Comment by roykho** [comment]
LGTM!

**Pull request merged**